platform/src/common/components/Collocation/DeviceStatus/ReportDetailPopup/index.jsx

```diff
@@ -1,10 +1,62 @@
 import { isEmpty } from 'underscore';
 import DetailCard from './detail_card';
 import { useRouter } from 'next/router';
+import Button from '@/components/Button';
+import { useEffect, useState } from 'react';
+import { useGetCollocationBatchResultsQuery } from '@/lib/store/services/collocation';
+import { saveAs } from 'file-saver';
+import ReactPDF, {
+  PDFViewer,
+  Document,
+  Page,
+  Text,
+  View,
+  PDFDownloadLink,
+  pdf,
+} from '@react-pdf/renderer';
+import ExportStatusReport from '../../BatchReport';
+import dynamic from 'next/dynamic';
+
+const DynamicPDFDownloadLink = dynamic(
+  () => import('@react-pdf/renderer').then((module) => module.PDFDownloadLink),
+  { ssr: false },
+);
 
 const ReportDetailCard = ({ deviceName, batchId, data, open, closeModal }) => {
   const router = useRouter();
-  // Adjust correlation & offset functionality
+  const [skip, setSkip] = useState(true);
+  const [collocationBatchId, setCollocationBatchId] = useState('');
+  const [exportData, setExportData] = useState([]);
+
+  const {
+    isLoading,
+    isError,
+    isSuccess,
+    data: collocationBatchResults,
+  } = useGetCollocationBatchResultsQuery(collocationBatchId, { skip: skip });
+  const collocationBatchResultsData = collocationBatchResults ? collocationBatchResults.data : null;
+  const [fileBlob, setFileBlob] = useState(null);
+
+  const downloadBatchReport = (batchId) => () => {
+    if (batchId !== '') {
+      setCollocationBatchId(batchId);
+      setSkip(false);
+    }
+  };
+
+  //   useEffect(() => {
```

Comment:
same here, let us remove this commented out code. @Codebmk

Reply:
The commented code is for the download report functionality which am still having trouble implementing @Baalmart